lib/ansible/plugins/lookup/first_found.py

```diff
@@ -195,15 +195,19 @@ def _process_terms(self, terms, variables, kwargs):
                         f = os.path.join(path, fn)
                         total_search.append(f)
             elif filelist:
-                # NOTE: this seems wrong, should be 'extend' as any option/entry can clobber all
-                total_search = filelist
+                # NOTE: this is now 'extend', previouslly it would clobber all options, but we deemed that a bug
+                total_search.extend(filelist)
```

Comment:
might be a bug but it was pre-existing behavior and some (sadly) seem to rely on it

Reply:
ok, I've removed this part for now. What error does this cause exactly? Is a separate backport required or documentation of this bug/change?